### ISSUE TITLE
CA2213 throwing error even when turned off in .editorconfig

### ISSUE THREAD

**Issue by dparvin** [dotnet/roslyn-analyzers#3486]
#### Analyzer package

[Microsoft.NetCore.Analyzers](https://www.nuget.org/packages/Microsoft.NetCore.Analyzers/)

[DisposableFieldsShouldBeDisposed](https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DisposableFieldsShouldBeDisposed.cs)

#### Package Version

v2.9.8 (latest release)

#### Diagnostic ID

[CA2213](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2213?view=vs-2019)

#### Repro steps

1. I have a DLL which contains a class with a variable in it that is getting disposed in the class's disposal routine.
2.I have a unit testing framework that creates a fakes version of the prior DLL with a bunch of parts of the original DLL not faked, but these specific parts are faked.
3.When I compile the Unit Testing framework DLL I get an error.

I have the CA2213 analyzer set to none in the .editorconfig file.
I am using Visual Studio 2019 Enterprise (16.5.2)
I am compiling with .NET 4.8

#### Expected behavior

I don't expect to see anything about CA2213

#### Actual behavior

I believe that the analyzer is looking at the fakes.dll file that is generated when making the fakes.dll.

I get an error like:

> Analyzer 'Microsoft.NetCore.Analyzers.Runtime.DisposableFieldsShouldBeDisposed' threw an exception of type 'System.ArgumentException' with message 'Reported diagnostic 'CA2213' has a source location in file 'xxxxxx', which is not part of the compilation being analyzed.

**Comment by Evangelink** [comment]
@dparvin Thank you for the feedback, could you please let us know if you are experiencing this issue in a consistent way or if it was a one-off? Do you have the nuget package and the VSIX extension in parallel? Are you experiencing the same behavior with the v3.0.0 of the package?

**Comment by Evangelink** [comment]
@dparvin ping

**Comment by dparvin** [comment]
Sorry, I have not tried it yet with 3.0.  I use the nuget packages to install the components.  I know I did something to get it to stop, if I remember correctly, I removed the nuget packages from the project that was causing the issue.  I have sense upgraded VS and reinstalled the nuget packages and I don't see the error.

**Comment by Evangelink** [comment]
@dparvin Thank you for the update! Feel free to reopen the ticket if you encounter again the issue.

**Comment by dparvin** [comment]
Ok, I have reproduced the problem again.  I had turned off the analyzers in the project throwing the error and so the error went away.  I recently upgraded from 2.9.8 to 3.0.0 and in the process figured out that I had disabled the Analyzers, when I turned the Analyzers back on, the error returned.  As I said before, I have a class in one DLL that does have a dispose function in it and there are variables in the class that need disposed and in that class, they are disposed.  

I have another DLL that runs tests in code and using Microsoft fakes, I have faked some of the functionality of the first class in the other DLL.  When this testing DLL is compiled, it makes a fakes version of the first DLL and I believe that fakes dll is what is getting Analyzed and giving the error.

Here is more of the information about the error.  I have not yet tried to reproduce this error in some other way where I can give you the code that reproduces it.  No matter what setting I put into the .editorconfig file, this error does not seem to go away.

> Error	AD0001	
> Analyzer 'Microsoft.NetCore.Analyzers.Runtime.DisposableFieldsShouldBeDisposed' threw an exception of type 'System.ArgumentException' with message 'Reported diagnostic 'CA2213' has a source location in file 'xxxxxxxxx', which is not part of the compilation being analyzed.
> Parameter name: diagnostic'.
> Exception occurred with following context:
> Compilation: yyyyyyy
> 
> System.ArgumentException: Reported diagnostic 'CA2213' has a source location in file 'xxxxxxxxx', which is not part of the compilation being analyzed.
> Parameter name: diagnostic
>    at Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalysisContextHelpers.VerifyDiagnosticLocationInCompilation(String id, Location location, Compilation compilation)
>    at Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalysisContextHelpers.VerifyDiagnosticLocationsInCompilation(Diagnostic diagnostic, Compilation compilation)
>    at Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalysisContextHelpers.VerifyArguments(Diagnostic diagnostic, Compilation compilationOpt, Func`2 isSupportedDiagnostic)
>    at Microsoft.CodeAnalysis.Diagnostics.SymbolAnalysisContext.ReportDiagnostic(Diagnostic diagnostic)
>    at Microsoft.NetCore.Analyzers.Runtime.DisposableFieldsShouldBeDisposed.<>c__DisplayClass8_0.<OnSymbolStart>g__OnSymbolEnd|7(SymbolAnalysisContext symbolEndContext)
>    at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c.<ExecuteSymbolEndActionsCore>b__53_1(ValueTuple`2 data)
>    at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
> -----
> 
> Suppress the following diagnostics to disable this analyzer: CA2213
>

**Comment by mavasani** [comment]
I believe this should be fixed in `Microsoft.CodeAnalysis.NetAnalyzers` version `5.0.3`
